owncloudApp/src/main/java/com/owncloud/android/presentation/ui/settings/fragments/SettingsSecurityFragment.kt

```diff
@@ -0,0 +1,218 @@
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+            if (result.resultCode != Activity.RESULT_OK) return@registerForActivityResult
+            val patternDisableResult = settingsViewModel.handleDisablePattern(result.data)
+            if (patternDisableResult.isSuccess) {
+                prefPattern?.isChecked = false
+
+                // Do not allow to use biometric lock since Pattern lock has been disabled
+                disableBiometric(getString(R.string.prefs_biometric_summary))
+            } else {
+                showMessageInSnackbar(getString(R.string.pattern_error_remove))
+            }
+        }
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.settings_security, rootKey)
+
+        screenSecurity = findPreference(SCREEN_SECURITY)
+        prefPasscode = findPreference(PassCodeActivity.PREFERENCE_SET_PASSCODE)
+        prefPattern = findPreference(PatternLockActivity.PREFERENCE_SET_PATTERN)
+        prefBiometric = findPreference(BiometricActivity.PREFERENCE_SET_BIOMETRIC)
+        prefTouchesWithOtherVisibleWindows = findPreference(PREFERENCE_TOUCHES_WITH_OTHER_VISIBLE_WINDOWS)
+
+        // Passcode lock
+        prefPasscode?.setOnPreferenceChangeListener { preference: Preference?, newValue: Any ->
+            if (settingsViewModel.isPatternSet()) {
+                showMessageInSnackbar(getString(R.string.pattern_already_set))
+            } else {
+                val intent = Intent(activity, PassCodeActivity::class.java)
+                if (newValue as Boolean) {
+                    intent.action = PassCodeActivity.ACTION_REQUEST_WITH_RESULT
+                    enablePasscodeLauncher.launch(intent)
+                } else {
+                    intent.action = PassCodeActivity.ACTION_CHECK_WITH_RESULT
+                    disablePasscodeLauncher.launch(intent)
+                }
+            }
+            false
+        }
+
+        // Pattern lock
+        prefPattern?.setOnPreferenceChangeListener { preference: Preference?, newValue: Any ->
+            if (settingsViewModel.isPasscodeSet()) {
+                showMessageInSnackbar(getString(R.string.passcode_already_set))
+            } else {
+                val intent = Intent(activity, PatternLockActivity::class.java)
+                if (newValue as Boolean) {
+                    intent.action = PatternLockActivity.ACTION_REQUEST_WITH_RESULT
+                    enablePatternLauncher.launch(intent)
+                } else {
+                    intent.action = PatternLockActivity.ACTION_CHECK_WITH_RESULT
+                    disablePatternLauncher.launch(intent)
+                }
+            }
+            false
+        }
+
+        // Biometric lock
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            screenSecurity?.removePreference(prefBiometric)
+        } else if (prefBiometric != null) {
```

Comment:
Is the `prefBiometric != null` needed? It should not be null at this point. You have some secure `?.` calls later anyway 🤔

Reply:
Well, it can be null if `findPreference` doesn't find it. It's true that we have some secure calls `?.` inside this else if, but this way, we save the `biometricManager = BiometricManager.getBiometricManager(activity)` instruction in case prefBiometric is null. Anyway, would it be worth it to transform the else if into else?

Reply:
Ok, I think that it will be null only when the `if condition is true`, but I'm not sure. If you think we need it, keep it, no problem 👍